wcomponents-core/src/main/java/com/github/bordertech/wcomponents/AbstractInput.java

```diff
@@ -296,24 +296,25 @@ protected boolean beforeHandleRequest(final Request request) {
 
 	/**
 	 * Perform change logic for this component.
+	 * <p>Reset focus ONLY if the current Request is an Ajax request. See https://github.com/BorderTech/wcomponents/issues/501.</p>
 	 */
 	protected void doHandleChanged() {
 		// If there is an associated action, execute it
 		if (getActionOnChange() != null) {
 			final ActionEvent event = new ActionEvent(this, getActionCommand(), getActionObject());
+			final WComponent currentInput = this;
 			Runnable later = new Runnable() {
 				@Override
 				public void run() {
 					getActionOnChange().execute(event);
-
-					if (isSubmitOnChange() && UIContextHolder.getCurrent().getFocussed() == null) {
+					if (AjaxHelper.isCurrentAjaxTrigger(currentInput) && UIContextHolder.getCurrent().getFocussed() == null) {
```

Comment:
What might be helpful is a new method in `AbstractInput` to check if the input is the current AJAX trigger. This could be useful for many input components that do this check....
```
     /**
	 *
	 * @return true if this input is the current AJAX trigger
	 */
	public boolean isCurrentAjaxTrigger() {
		return AjaxHelper.isCurrentAjaxTrigger(this);
	}
```

Also noticed `WTab` has a static method `isCurrentAjaxTrigger`. Might be good to deprecate this and call the AjaxHelper method.